Feature/Task-12 Scalfolded models, entity classes were created

2musime opened the pull request:
## Entity Classes for Certificates, Users, Suppliers, and Comments using the Db-First Approach
### Description
This PR introduces the entity classes for the Certificate, User, Supplier, and Comment tables, generated using the Db-First approach from the `certificatedb` MS SQL Server database. Entity classes have been scaffolded and mapped to the database schema of application models.
### Changes:


- [x] Scaffolded models for Certificates, Users, Suppliers, and Comments tables.
- [x] Connection string used for scaffolding.
- [x] Models are located in the Entities directory.

2musime committed:
Feature/Task-12 Scalfolded models, entity classes were created

2musime committed:
Feature/Task-12 Named correctly to my entities

elhasic commented on backend/CertificateManagerAPIs/Entities/CertificatedbContext.cs:
```diff
@@ -0,0 +1,105 @@
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see https://go.microsoft.com/fwlink/?LinkId=723263.
+        => optionsBuilder.UseSqlServer("Data Source=TUMUSIME;Initial Catalog=certificatedb;Integrated Security=True;Connect Timeout=30;Encrypt=True;Trust Server Certificate=True;Application Intent=ReadWrite;Multi Subnet Failover=False");
```
Please remove this connection string source, because you use connections string from appsettings

elhasic commented:
Please add folder "Scripts" where you will add all your DB SQL scripts by date ordering, and also add folders "Tools" where you will put your Scaffold script

2musime committed:
Feature/Task-12 Created Scripts to keep DB SQL scripts and Tools to put Scaffold script

elhasic commented on backend/CertificateManagerAPIs/CertificateManagerAPIs.csproj:
```diff
@@ -13,13 +13,14 @@
     </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="9.0.0-rc.1.24451.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.8" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="9.0.0-rc.1.24451.1">
```
please do not use RC versions of packages

elhasic commented on backend/CertificateManagerAPIs/Data/CertificatedbContext.cs:
```diff
@@ -0,0 +1,105 @@
+
+    public virtual DbSet<User> Users { get; set; }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
This method should not be here, all infos regarding connection string prepare on appsettings in program.cs

2musime replied:
Resolved.

2musime committed:
- Created new database SQL Script
 - Runned Scalfolding Scripts
 - New Models were created based on the database structure.
 - Fixed the versions issues by installing latest stable versions.
 - Moved appsetting.Development to git ignore.

2musime committed:
Feature/Task-12 Correct connection string from public files hence the separation of concern

elhasic commented on backend/CertificateManagerAPIs/Services/ServiceExtension.cs:
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.EntityFrameworkCore;
+namespace CertificateManagerAPIs.Services
+{
+    public static class ServiceExtension
```
This is not service, this is extension method for connection string configuration, so should not be in Services folder

2musime replied:
This is also solved.
Thank you.

2musime committed:
Feature/Task-12 Properly located connection string and deleted unwanted files
